_episodes/01-raster-structure.md

```diff
@@ -15,10 +15,474 @@ objectives:
+regardless of whether it has one or more bands and places each in a third 
+axis of a numpy array in `[bands, rows, columns]` order. For example, even if 
+there is only 1 band in a raster, it will be placed in it's own band axis.
+
+~~~
+surface_model_HARV_arr_3D = surface_model.read()
+surface_model_HARV_arr_3D.shape
+~~~
+{: .language-python}
+
+
+
+~~~
+(1, 1367, 1697)
+~~~
+{: .output}
+
+Earlier, we read our raster in as a 2D array instead
+
+~~~
+surface_model_HARV_arr_2D = surface_model.read(1)
+surface_model_HARV_arr_2D.shape
+~~~
+{: .language-python}
+
+
+
+~~~
+(1367, 1697)
+~~~
+{: .output}
+
+It's always a good idea to examine the shape of the raster array you are working with and make sure it's what you expect. Many functions, especially ones that plot images, expect a raster array to have a particular shape.
+
+Jump to a later episode in
+this series for information on working with multi-band rasters:
+[Work with Multi-band Rasters in Python]({{ site.baseurl }}/05-raster-multi-band/).
+
+## Dealing with Missing Data
+
+Raster data often has a no data value associated with it. This is a value
+assigned to pixels where data is missing or no data were collected. However, 
+there can be different cases that cause missing data, and it's common for other 
+values in a raster to represent different cases. The most common example is missing 
+data at the edges of rasters.
+
+By default the shape of a raster is always rectangular. So if we have a dataset
+that has a shape that isn't rectangular, some pixels at the edge of the raster
+will have no data values. This often happens when the data were collected by an
+sensor which only flew over some part of a defined region.
+
+In the RGB image below, the pixels that are black have no data values. The sensor
+did not collect data in these areas.
+
+~~~
+rgb_HARV = rasterio.open("NEON-DS-Airborne-Remote-Sensing/HARV/RGB_Imagery/HARV_RGB_Ortho.tif", "r")
+rgb_HARV_arr = rgb_HARV.read()
+earthpy.plot.plot_rgb(
+    rgb_HARV_arr,
+    title="RGB Image, NoData Values UnMasked",
+    figsize=(10, 6)
+)
+~~~
+{: .language-python}
+
+
+<img src="../fig/01-demonstrate-no-data-black-3.png" title="plot of chunk demonstrate-no-data-black" alt="plot of chunk demonstrate-no-data-black" width="612" style="display: block; margin: auto;" />
+
+`rasterio` assigns a specific number as missing data to the `meta` attribute when the dataset is read, base don the file's own metadata. While the GeoTiff's `nodata` attribute is assigned to the value `-1.7e+308`, it turns out the missing data at the edges are represented by the value `0`. In order to run calculations on this image that ignore these edge values or plot he iamge without the nodata values being displayed on the color scale, we can mask out `0` values in our numpy array. 
+
+In the next image, the black edges have been masked using `numpy.ma.masked_where()`, a function that assigns no data values where a condition is true.
+
```

Comment:
You use numpy.ma here (and imported numpy), but then below use np.ma.masked_where()... We should probably check for consistency later.

Reply:
Made this consistent, thanks for the catch.